src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -332,38 +334,101 @@ int NonClientIslandWindow::_GetResizeHandleHeight() const noexcept
+    // WS_POPUP size, so we don't have to worry about borders, and the default
+    // frame will be fine.
+    if (_isMaximized && !_fullscreen)
     {
         // When a window is maximized, its size is actually a little bit more
         // than the monitor's work area. The window is positioned and sized in
         // such a way that the resize handles are outside of the monitor and
         // then the window is clipped to the monitor so that the resize handle
         // do not appear because you don't need them (because you can't resize
         // a window when it's maximized unless you restore it).
-        newTop += _GetResizeHandleHeight();
+        newSize.top += _GetResizeHandleHeight();
+    }
+
+    // GH#1438 - Attempt to detect if there's an autohide taskbar, and if there
+    // is, reduce our size a bit on the side with the taskbar, so the user can
+    // still mouse-over the taskbar to reveal it.
+    HMONITOR hMon = MonitorFromWindow(_window.get(), MONITOR_DEFAULTTONULL);
+    if (hMon && (_isMaximized || _fullscreen))
+    {
+        MONITORINFO monInfo{ 0 };
+        monInfo.cbSize = sizeof(MONITORINFO);
+        GetMonitorInfo(hMon, &monInfo);
+
+        // First, check if we have an auto-hide taskbar at all:
+        APPBARDATA autohide{ 0 };
+        autohide.cbSize = sizeof(autohide);
+        UINT state = (UINT)SHAppBarMessage(ABM_GETSTATE, &autohide);
+        if (WI_IsFlagSet(state, ABS_AUTOHIDE))
+        {
+            // This helper can be used to determine if there's a auto-hide
+            // taskbar on the given edge of the monitor we're currently on.
+            auto hasAutohideTaksbar = [&monInfo](const UINT edge) -> bool {
+                APPBARDATA data{ 0 };
+                data.cbSize = sizeof(data);
+                data.uEdge = edge;
+                data.rc = monInfo.rcMonitor;
+                HWND hTaskbar = (HWND)SHAppBarMessage(ABM_GETAUTOHIDEBAREX, &data);
+                return hTaskbar != nullptr;
+            };
+
+            const bool onTop = hasAutohideTaksbar(ABE_TOP);
+            const bool onBottom = hasAutohideTaksbar(ABE_BOTTOM);
+            const bool onLeft = hasAutohideTaksbar(ABE_LEFT);
+            const bool onRight = hasAutohideTaksbar(ABE_RIGHT);
+
+            // If there's a taskbar on any side of the monitor, reduce our size
+            // a little bit on that edge.
+            //
+            // Note to future code archeologists:
+            // This doesn't seem to work for fullscreen on the primary display.
+            // However, testing a bunch of other apps with fullscreen modes
+            // and an auto-hiding taskbar has shown that _none_ of them
+            // reveal the taskbar from fullscreen mode. This includes Edge,
+            // Firefox, Chrome, Sublime Text, Powerpoint - none seemed to
+            // support this.
+            //
+            // This does however work fine for maximized.
+            if (onTop)

```

Comment:
nit: a taskbar can only be on one side of a monitor at a time right? then these if-statements could turn into if-elses

Reply:
Actually, _no_. It appears that apps can use this API to add additional bars to other sides of the display? It's really unclear, this whole API is a disaster -.-

Reply:
wtf yikes 😨